### PR TITLE
chore(sdk): adding bucket sdk tests

### LINKED ISSUE #2727: Adding bucket sdk tests
### Feature Spec

The test will be written in Wing.
it will cover the following methods of the Bucket resource:

 List
 - [ ] AddObject, Delete, List
 - [ ] Public
 - [ ] PublicUrl
 
Please be minded to test the methods with all possible arguments compositions, and edge cases ✨
The tests should be placed under `examples/tests/sdk_tests/bucket`

### Use Cases

testing

### Implementation Notes

_No response_

### Component

SDK

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

### PR DESCRIPTION
**Pull request by garysassano**
Closes #2727

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: examples/tests/sdk_tests/bucket/delete.w [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+test "delete" {
+  let jsonObj1 = Json { key1: "value1" };
+
+  b.putJson("file1.json", jsonObj1);
+  b.delete("file1.txt");
+
+  assert(b.list().length == 1);
+
+  b.delete("file1.json");
+  b.delete("file1.json");
```
Comment (tsuf239) [inline examples/tests/sdk_tests/bucket/delete.w:14]: can you please test with the `mustExist` argument too? (and deleting a file uploaded via `addObject`?)

File: examples/tests/sdk_tests/bucket/list.w [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  assert(b.list().length == 0);
+
+  b.put("file1.txt", "Foo");
+  b.put("file2.txt", "Bar");
+
+  testArray = b.list().copyMut();
+
+  assert(b.get(testArray.at(0)) == "Foo");
+  assert(b.get(testArray.at(1)) == "Bar");
```
Comment (tsuf239) [inline examples/tests/sdk_tests/bucket/list.w:16]: I think testing the keys is enough (otherwise we'll actually test `b.get`) :) 

you can take a look here:
examples/tests/valid/bucket_keys.w
https://github.com/winglang/wing/blob/0ff799d689555f697c0ffcacf0aae48c2f53bf5e/examples/tests/valid/bucket_keys.w#L1-L17

File: examples/tests/sdk_tests/bucket/list.w [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+let b = new cloud.Bucket();
+
+test "list" {
+  let var testArray = MutArray<str>[];
+
+  assert(b.list().length == 0);
+
+  b.put("file1.txt", "Foo");
+  b.put("file2.txt", "Bar");
```
Comment (tsuf239) [inline examples/tests/sdk_tests/bucket/list.w:11]: can you please check if `list` works with `putJson` and `addObject`?

File: examples/tests/sdk_tests/bucket/list.w [hunk 4]
```diff
@@ -0,0 +1,21 @@
+bring cloud;
+
+let b = new cloud.Bucket();
+
+test "list" {
```
Comment (tsuf239) [inline examples/tests/sdk_tests/bucket/list.w:5]: can you also test `list` with `prefix` argument?

File: examples/tests/sdk_tests/bucket/public_url.w [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  privateBucket.put("file2.txt", "Bar");
+
+  assert(publicBucket.publicUrl("file1.txt") != "");
+
+  try {
+    privateBucket.publicUrl("file2.txt");
+  } catch e {
+    assert(e == "Cannot provide public url for a non-public bucket");
+  }
+}
```
Comment (tsuf239) [inline examples/tests/sdk_tests/bucket/public_url.w:18]: good!
(soon we'll have fetch and then we can test if by sending a GET request to the public URL, we actually get the value)
I'll open an issue to add to this one :)

File: examples/tests/sdk_tests/bucket/delete.w [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+let b = new cloud.Bucket();
+b.addObject("file3.txt", "Baz");
+
+test "delete" {
+  let jsonObj1 = Json { key1: "value1" };
+
+  b.putJson("file1.json", jsonObj1);
+  b.delete("file1.txt");
+
+  assert(b.list().length == 2);
```
Comment (tsuf239) [inline examples/tests/sdk_tests/bucket/delete.w:12]: maybe check with `exists` instead of `list`?

File: examples/tests/sdk_tests/bucket/delete.w [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+  b.delete("file1.json", { mustExist: true });
+  b.delete("file1.json", { mustExist: false });
```
Comment (tsuf239) [inline examples/tests/sdk_tests/bucket/delete.w:15]: can you please catch and assert the error?

File: examples/tests/sdk_tests/bucket/list.w [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  let objs = b.list();
+  let objs2 = b.list("file");
+
+  assert(objs.at(0) == "file1.txt");
+  assert(objs.at(1) == "file2.txt");
+  assert(objs.at(2) == "random");
+
+  assert(objs2.at(0) == "file1.txt");
+  assert(objs2.at(1) == "file2.txt");
+  assert(!(objs2.at(2) == "random"));
```
Comment (tsuf239) [inline examples/tests/sdk_tests/bucket/list.w:21]: maybe use `contains` ?

File: examples/tests/sdk_tests/bucket/list.w [hunk 9]
```diff
@@ -0,0 +1,25 @@
+bring cloud;
```
Comment (tsuf239) [inline examples/tests/sdk_tests/bucket/list.w:1]: The files are saved to the snapshots only by their file name, regarding the path... we already have a "table/list" can you please change this file name to "bucket_list" until we'll fix that issue? (#2781)

File: examples/tests/sdk_tests/bucket/delete.w [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+  b.delete("file1.txt");
+
+  assert(b.exists("file1.json"));
+  assert(b.exists("file2.txt"));
+
+  b.delete("file1.json", { mustExist: true });
+  try {
+    b.delete("file1.json", { mustExist: true });
+  } catch e {
+    assert(e == "Object does not exist (key=file1.json).");
```
Comment (tsuf239) [inline examples/tests/sdk_tests/bucket/delete.w:19]: please note that if there's no error, this assert won't be called...

**Review by tsuf239 — APPROVED** [review]
Amazing! thanks for the fixes!

File: examples/tests/sdk_tests/bucket/bucket_list.w [hunk 11]
```diff
@@ -0,0 +1,27 @@
+bring cloud;
+
+let b = new cloud.Bucket();
+
+test "list" {
+  let jsonObj1 = Json { key1: "value1" };
```
Comment (tsuf239) [inline examples/tests/sdk_tests/bucket/bucket_list.w:6]: can you please add something via `addObject` to make sure it's listed too?
